components/post_view/combined_system_message/combined_system_message.tsx

```diff
@@ -238,30 +238,40 @@ export class CombinedSystemMessage extends React.PureComponent<Props> {
         return usernames;
     }
 
-    getUsernamesByIds = (userIds: string[] = []): string[] => {
+    getUsernamesByIds = (userIds: string | string[] = []): string[] => {
+        const userIdsArray = Array.isArray(userIds) ? userIds : [userIds];
         const {currentUserId, currentUsername} = this.props;
         const allUsernames = this.getAllUsernames();
 
         const {formatMessage} = this.props.intl;
         const someone = formatMessage({id: t('channel_loader.someone'), defaultMessage: 'Someone'});
 
-        const usernames = userIds.
+        const usernames = userIdsArray.
             filter((userId) => {
                 return userId !== currentUserId && userId !== currentUsername;
             }).
             map((userId) => {
                 return allUsernames[userId] ? `@${allUsernames[userId]}` : someone;
-            }).filter((username) => {
+            }).
+            filter((username) => {
                 return username && username !== '';
             });
 
-        if (userIds.includes(currentUserId)) {
+        if (userIdsArray.includes(currentUserId)) {
             usernames.unshift(allUsernames[currentUserId]);
-        } else if (userIds.includes(currentUsername)) {
+        } else if (userIdsArray.includes(currentUsername)) {
             usernames.unshift(allUsernames[currentUsername]);
         }
 
-        return usernames;
+        // check for duplicates in usernames
```

Comment:
Didn't we already check for duplicates in `extractUserData`?

If we do need this though, I think it's quicker if we use a `Set` to remove duplicates than to construct a new array by recreating it while checking if we've already added the item. Sets in JS are ordered by the order you insert items, so this is a quicker way of writing that

```js
Array.from(new Set(usernames));
```